stan/math/gpu/kernels/check_diagonal_zeros_kernel.cl

```diff
@@ -0,0 +1,16 @@
+R"(
+#ifndef A
+#define A(i, j)  A[j * rows + i]
+#endif
+__kernel void check_diagonal_zeros(
```

Comment:
can we rename these functions? in the math library, we have a distinction between `is_*` and `check_*` where `is_*` returns a bool and `check_*` throws. 

Since this is on the kernel, it can be called anything, but if you don't mind, could we rename it so it doesn't conflict with the naming convention of the math library?

Reply:
I have renamed to is_nan, is_symmetric (also flipped the flag) and is_zero_on_diagonal. If you would prefer other names, feel free to suggest.